### PR TITLE
Tasks 970-1086: sherliic, prachathai67k and pib

### PR DESCRIPTION
**Pull request by CodeHime**

### CODE REVIEW

**Comment by swarooprm** [comment]
Reviews:https://github.com/swarooprm/CSE576HW/pull/105

**Comment by swarooprm** [comment]
@Mehrad0711, wondering if you can help review this PR since you have a lot of experience in creating translation tasks?

**Comment by Mehrad0711** [comment]
The task structure looks good. I can't comment on the text cause I can't read them! 
One suggestion is adding more descriptive explanations for negative examples (i.e. WHY they are incorrect).

**Comment by CodeHime** [comment]
> The task structure looks good. I can't comment on the text cause I can't read them! One suggestion is adding more descriptive explanations for negative examples (i.e. WHY they are incorrect).

The translations hold a different meaning from the input sentences so they are incorrect. Would it work if I added this line to the explanations?

**Comment by Mehrad0711** [comment]
> The translations hold a different meaning from the input sentences so they are incorrect. Would it work if I added this line to the explanations?

I thought in some cases you have a correct translation from which you've omitted a word, or have changed the order of phrases, etc. to make it incorrect. If in all negative examples, the translation is completely irrelevant, no need to.

**Comment by CodeHime** [comment]
Okay, since the translation is completely irrelevant, I will keep the tasks as they are.

**Comment by swarooprm** [comment]
> Okay, since the translation is completely irrelevant, I will keep the tasks as they are.

Any update here @CodeHime ?

**Comment by CodeHime** [comment]
> > Okay, since the translation is completely irrelevant, I will keep the tasks as they are.
> 
> Any update here @CodeHime ?

HI,
Let me know if I misunderstood something, but since the translations are irrelevant, I was told there was no need to make any changes here. Please confirm @Mehrad0711 @swarooprm

**Comment by swarooprm** [comment]
> > > Okay, since the translation is completely irrelevant, I will keep the tasks as they are.
> > 
> > 
> > Any update here @CodeHime ?
> 
> HI, Let me know if I misunderstood something, but since the translations are irrelevant, I was told there was no need to make any changes here. Please confirm @Mehrad0711 @swarooprm

Ok, thanks for clarifying.

**Comment by CodeHime** [comment]
> > > > Okay, since the translation is completely irrelevant, I will keep the tasks as they are.
> > > 
> > > 
> > > Any update here @CodeHime ?
> > 
> > 
> > HI, Let me know if I misunderstood something, but since the translations are irrelevant, I was told there was no need to make any changes here. Please confirm @Mehrad0711 @swarooprm
> 
> Ok, thanks for clarifying.

My pleasure. Would I need to make any other changes here? Or is this complete?

**Comment by swarooprm** [comment]
> Okay, since the translation is completely irrelevant, I will keep the tasks as they are.



> > > > > Okay, since the translation is completely irrelevant, I will keep the tasks as they are.
> > > > 
> > > > 
> > > > Any update here @CodeHime ?
> > > 
> > > 
> > > HI, Let me know if I misunderstood something, but since the translations are irrelevant, I was told there was no need to make any changes here. Please confirm @Mehrad0711 @swarooprm
> > 
> > 
> > Ok, thanks for clarifying.
> 
> My pleasure. Would I need to make any other changes here? Or is this complete?

I will get back to you. @aarunku5 can you have a second look? There are so many tasks here, so it would be better if you also review it

**Comment by aarunku5** [comment]
Sure, can you fix the tasks 970-976 first, and I can check those corrections while you work on the translation task changes?

970 - You don't need to say "sentence1 and sentence2" in the definition. Also add a sentence to define what 'entail' means. Grammatical error in your explanation for the second negative example. Also did you mean the task name to be "causal" and not "casual" relationship.

971- Can you reword the definition slightly to make it clearer? Say that there are two placeholders or spaces or something of the sort for the objects in the fact, for which examples are provided. It's not fully clear on just reading the definition. Also, can you clarify what you mean by "annotation is correct"? In your positive examples, for the first one, for example, if [A] is Portugal and [B] is the UK, then the annotation is incorrect. How do you ascertain the annotation correctness? Is there a 1-1 correspondence in the examples listed for [a] and [b]? Also check if there are such cases in the instances as well.

972- Typo, "detemine". "Follow the following rules while answering the order"- rephrase this. To clarify- "If A and B are premise-hypothesis pairs of each other"- by this you mean that "AB" and "BA" are both correct? Can you reword the rules to be a bit more concise? Typos and grammatical errors in the examples and their explanations as well.

973- Typo, "detemine". Typos and grammatical errors in the examples and their explanations as well.

974- In the definition, you need to specify the language of the input text, and that the genre is specified in English. In the examples and instances, split the input into 2 fields- input: <passage>, genre: <genre>. You don't need to ask a full question "What is the sentiment ...", just have a genre field instead.  Explanations: Typo "tallks". Say "about the genre" instead of "for the genre".

975- Similar to 974, get rid of "Do both the articles share the same sentiment for the genre: " and have 3 fields- article 1, article 2, genre. Change the definition: it's 2 articles, not 2 sentences. And specify the language of the articles. Also, in your definition you say same genre, but your examples indicate you are checking if they are speaking with the same sentiment? I'm assuming you mean sentiment checking in the definition as well (if so update that)? For your explanations, split it into 2 sentences-->  "Article 1  talks positively about the genre culture. Article 2......"

976- Typo "onf". Grammatical errors in definition. 

For all translation tasks (977-1086):
1. Your positive and negative example explanations must be more detailed. What do the two sentences say, and why is it a correct/incorrect translation?
2. Specify in the definition that the translation must not omit or add information wrt the original sentence.
3. Some of your instances and examples have multiple sentences in the input and output. So change the definition to say you are given an "input text", and not a sentence.
4. Can you just verify that on translation, any numbers in your input text are either translated to number representations or numbers in word form in the translated text?

**Comment by CodeHime** [comment]
> Sure, can you fix the tasks 970-976 first, and I can check those corrections while you work on the translation task changes?
> 
> 970 - You don't need to say "sentence1 and sentence2" in the definition. Also add a sentence to define what 'entail' means. Grammatical error in your explanation for the second negative example. Also did you mean the task name to be "causal" and not "casual" relationship.
> 
> 971- Can you reword the definition slightly to make it clearer? Say that there are two placeholders or spaces or something of the sort for the objects in the fact, for which examples are provided. It's not fully clear on just reading the definition. Also, can you clarify what you mean by "annotation is correct"? In your positive examples, for the first one, for example, if [A] is Portugal and [B] is the UK, then the annotation is incorrect. How do you ascertain the annotation correctness? Is there a 1-1 correspondence in the examples listed for [a] and [b]? Also check if there are such cases in the instances as well.
> 
> 972- Typo, "detemine". "Follow the following rules while answering the order"- rephrase this. To clarify- "If A and B are premise-hypothesis pairs of each other"- by this you mean that "AB" and "BA" are both correct? Can you reword the rules to be a bit more concise? Typos and grammatical errors in the examples and their explanations as well.
> 
> 973- Typo, "detemine". Typos and grammatical errors in the examples and their explanations as well.
> 
> 974- In the definition, you need to specify the language of the input text, and that the genre is specified in English. In the examples and instances, split the input into 2 fields- input: , genre: . You don't need to ask a full question "What is the sentiment ...", just have a genre field instead. Explanations: Typo "tallks". Say "about the genre" instead of "for the genre".
> 
> 975- Similar to 974, get rid of "Do both the articles share the same sentiment for the genre: " and have 3 fields- article 1, article 2, genre. Change the definition: it's 2 articles, not 2 sentences. And specify the language of the articles. Also, in your definition you say same genre, but your examples indicate you are checking if they are speaking with the same sentiment? I'm assuming you mean sentiment checking in the definition as well (if so update that)? For your explanations, split it into 2 sentences--> "Article 1 talks positively about the genre culture. Article 2......"
> 
> 976- Typo "onf". Grammatical errors in definition.
> 
> For all translation tasks (977-1086):
> 
> 1. Your positive and negative example explanations must be more detailed. What do the two sentences say, and why is it a correct/incorrect translation?
> 2. Specify in the definition that the translation must not omit or add information wrt the original sentence.
> 3. Some of your instances and examples have multiple sentences in the input and output. So change the definition to say you are given an "input text", and not a sentence.
> 4. Can you just verify that on translation, any numbers in your input text are either translated to number representations or numbers in word form in the translated text?

Sure. Let me get back to you as soon as I can (either EOD or latest tomorrow) on these updates.

**Comment by CodeHime** [comment]
Hi @aarunku5 
I just want to know if the input can be split into multiple parts in any given task?
I mean, can I split input in task970 into sentence1 and sentence2 as well? Can I do this with any other task I have submitted?

**Comment by aarunku5** [comment]
Yes you can do that. Your input can consist of multiple fields for any task, as long as all the fields are relevant and are used to solve the task.

**Comment by aarunku5** [comment]
@CodeHime any updates from your end?

**Comment by CodeHime** [comment]
> @CodeHime any updates from your end?

The translation tasks are pending to be updated:

> Your positive and negative example explanations must be more detailed. What do the two sentences say, and why is it a correct/incorrect translation?
I added this but it seems that google translate is not able to handle translation of long sentences properly. So working on that,

> Can you just verify that on translation, any numbers in your input text are either translated to number representations or numbers in word form in the translated text?
The number format is converted into either one of both number representations and word representations .

**Comment by aarunku5** [comment]
All right, let me know once you're ready for review.

**Comment by CodeHime** [comment]
> All right, let me know once you're ready for review.

Hi
I need to manually translate and verify all 100 translation tasks. I will need one or two more days. I will let you know as soon as it completes.

**Comment by danyaljj** [comment]
Please make sure to set the `"Categories": [""]` and `"Domains": [""]` fields according to [the taxonomy presented here](https://github.com/allenai/natural-instructions-expansion/blob/master/doc/task-hierarchy.md).

**Comment by CodeHime** [comment]
Please check. Also, the "input" field is deemed necessary in the test checks so I have re-structured accordingly.

File: tasks/task971_sherliic_fact_annotation.json [hunk 1]
```diff
@@ -0,0 +1,15823 @@
...
+      "Categories": [
+            "Fact Annotation"
+      ],
+      "Domains": [
+            "Formal logic"
+      ],
+      "Definition": "In this task, you will be given a fact as input, a possible annotation and some examples. The fact will always contain exactly two placeholders [A] and [B], with some examples for both. You will also be given a possible annoation of the fact. Assume that the fact holds true for the given examples. Tell if, given, the fact is true, the annotation is also true. If the annotation holds true, answer \"yes\", \"no\" or \"incomprehensible\". ",
+      "Positive Examples": [
+            {
+                  "input": "fact:region[A] is nation in location[B]\nexamples[A]:Portugal / England / Tanzania\nexamples[B]:United Kingdom / Central America / Europe\nannotation:region[A] is country in location[B]",
```
Comment (danyaljj) [inline tasks/task971_sherliic_fact_annotation.json]: The phrasing of the input is quite difficult to parse for an average human: 
```
fact:region[A] is nation in location[B]\nexamples[A]:Portugal / England / Tanzania\nexamples[B]:United Kingdom / Central America / Europe\nannotation:region[A] is country in location[B]
```
Like, here, I am not sure what `\nannotation:region[A] is country in location[B]` at the end of the example is saying. 


Wondering if we can simplify it somehow? 
Here is an example (and I hope it doesn't compromise any details): 

```
In following relationship, [region] is nation in [location], examples of [region] can be 'Portugal', 'England', 'Tanzania' and examples of [location] can be 'United Kingdom', 'Central America' 'Europe'.  
```

**Comment by danyaljj** [comment]
Looks like one of the tests are failing as well.

**Comment by CodeHime** [comment]
> Looks like one of the tests are failing as well.

Are the input languages to be a list of strings in translation tasks?

**Comment by danyaljj** [comment]
> > Looks like one of the tests are failing as well.
> 
> Are the input languages to be a list of strings in translation tasks?

Not sure if I understand.

**Comment by swarooprm** [comment]
> > Looks like one of the tests are failing as well.
> 
> Are the input languages to be a list of strings in translation tasks?

Yes, they need to be a list: Some thing like this:
"Input_language": [
        "English"
    ],
    "Output_language": [
        "English"
    ],
    "Instruction_language": [
        "English"
    ],

File: tasks/task971_sherliic_fact_annotation.json [hunk 2]
```diff
@@ -0,0 +1,15832 @@
...
+      "Input_language": [
+        "English"
+      ],
+      "Output_language": [
+        "English"
+      ],
+      "Instruction_language": [
+        "English"
+      ],
+      "Definition": "In this task, you will be given a fact as input, a possible annotation and some examples. The fact will always contain exactly two placeholders [A] and [B], with some examples for both. You will also be given a possible annoation of the fact. Assume that the fact holds true for the given examples. Tell if, given, the fact is true, the annotation is also true. If the annotation holds true, answer \"yes\", \"no\" or \"incomprehensible\". ",
```
Comment (danyaljj) [inline tasks/task971_sherliic_fact_annotation.json]: Grammar issues here: "Tell if, given, the fact is true, the annotation is also true.""

File: tasks/task971_sherliic_fact_annotation.json [hunk 3]
```diff
@@ -0,0 +1,15832 @@
...
+                  ]
+            },
+            {
+                  "input": "fact:person[A] is elected to be President of region[B] \nexamples[A]:Nguy\u1ec5n V\u0103n Thi\u1ec7u / William Henry Harrison / John Quincy Adams\nexamples[B]:United States of America\nannotation:person[A] is elected president of region[B] ",
+                  "output": [
+                        "yes"
+                  ]
+            },
+            {
+                  "input": "fact:organization_member[A] is aligning with organization_founder[B] \nexamples[A]:Silvio Berlusconi / Canada / Egypt\nexamples[B]:United States of America / France / Gianfranco Fini\nannotation:organization_founder[B] is   organization_member[A] 's partner",
```
Comment (danyaljj) [inline tasks/task971_sherliic_fact_annotation.json]: There are three examples for [A] and three examples for [B].
How should one read these examples? for instance, is there a 1-to-1 correspondance between pairs of the examples?
 - (A, B) = (Silvio Berlusconi, United States of America)
 - (A, B) = (Canada, France)
 - (A, B) = (Egypt, Gianfranco)

Comment (CodeHime) [reply tasks/task971_sherliic_fact_annotation.json]: There isn't a one to one correspondence. They are given as given in the dataset. I made no changes to them.
 
They are simply values that can fill A and B. They need not make the statement true. e.g.  Silvio Berlusconi / Canada / Egypt are of type organization_members so they can fill A. United States of America / France / Gianfranco Fini are of type organization_founder so they can fill B. A and B are not pairs that make the fact true, just examples of what can fill A and B.

Comment (danyaljj) [reply tasks/task971_sherliic_fact_annotation.json]: In that case, how about changing the word to "a, b, c, etc." 
 - Slashes are not common in written English 
 - "etc." to emphasize that these are samples of potentially many examples.

Comment (CodeHime) [reply tasks/task971_sherliic_fact_annotation.json]: Okay

Comment (danyaljj) [reply tasks/task971_sherliic_fact_annotation.json]: Also, I think you can simply the inputs, if you drop [A] and [B] and put the brackets around the two nodes.

Comment (CodeHime) [reply tasks/task971_sherliic_fact_annotation.json]: Just need a clarification: I was simplifying the input and was suggested to split the input into fact, examples and annotation. This means that convert the input into a dictionary right? Like 
"input": {"fact":"Some fact here", "examples":"some examples, etc", "annotation":"some annotation"}

Comment (danyaljj) [reply tasks/task971_sherliic_fact_annotation.json]: No, you can still maintain text inputs, I think. I'm suggesting a small simplification in how the input is phrased.

**Comment by danyaljj** [comment]
Dropping the following tasks so that we can get this PR in. 
 - 971, 972, 973: the task is not unclear and needs major revision before we can get it in. 
Happy to reconsider these tasks, if anyone is interested in improving them and sending these in a different PR.

**Pull request merged**